solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java

```diff
@@ -91,6 +91,12 @@
 import static org.apache.solr.common.params.CommonParams.ADMIN_PATHS;
 import static org.apache.solr.common.params.CommonParams.ID;
 
+/**
+ * If you don't call the {@link #setIdField(String)} method and
+ * if your collection's document router has no route field then
+ * this class assumes the id field for your documents is called
+ * 'id'.
+ */
```

Comment:
This is relocated (to avoid duplication) from the two deriving classes and then reworded to try and make the context for the deprecated `setIdField` method clearer. Other than this usage I was unable to identify anywhere else where the cloud solr client assumes that the id field is called `"id"`. What do you think?

Reply:
Javadocs, particularly class javadocs, should start with a sentence about what this class is / is-for.  The javadoc you have here describes some details that is relevant but not something to *start* javadoc with.  So please add a sentence like "Base SolrClient for implementations optimized for SolrCloud.

Reply:
Hmm, yes, that's a fair point. Also the precommit checks don't like the `{@inheritDoc}` use in the derived class somehow (though elsewhere in the code base it is used) and the reworded content here does not meet the initial objective of the pull request which was to _not_ mention the deprecated `setIdField` method in the javadocs. Oh well, it was worth a try. Thanks @dsmiley for your input!